src/twister2/device/native_simulator_adapter.py

```diff
@@ -29,29 +29,41 @@ class NativeSimulatorAdapter(DeviceAbstract):
     """Adapter class for a device simulator."""
 
     def __init__(self, twister_config: TwisterConfig, **kwargs) -> None:
+        """
+        :param twister_config: twister configuration
+        """
         super().__init__(twister_config, **kwargs)
         self._process: asyncio.subprocess.Process | None = None
         self._process_ended_with_timeout: bool = False
         self.queue: Queue = Queue()
         self._stop_job: bool = False
         self._exc: Exception | None = None  #: store any exception which appeared running this thread
         self._thread: threading.Thread | None = None
-
-    @staticmethod
-    def _get_command(build_dir: Path | str) -> list[str]:
+        self.command: list[str] = []
+        self.process_kwargs: dict = {
+            'stdout': asyncio.subprocess.PIPE,
+            'stderr': asyncio.subprocess.STDOUT,
+            'env': self.env,
+        }
+
+    def generate_command(self, build_dir: Path | str) -> list[str]:
         """
         Return command to run.
 
         :param build_dir: build directory
         :return: command to run
         """
-        return [str((Path(build_dir) / 'zephyr' / 'zephyr.exe').resolve())]
+        self.command = [str((Path(build_dir) / 'zephyr' / 'zephyr.exe').resolve())]
 
     def connect(self, timeout: float = 1) -> None:
         pass
 
-    def run(self, build_dir: str | Path, timeout: float = 60.0) -> None:
-        self._thread = threading.Thread(target=self._run_simulation, args=(build_dir, timeout), daemon=True)
+    def flash_and_run(self, timeout: float = 60.0) -> None:
+        if self.command == []:
+            msg = 'Run simulation command is empty, please verify if it was generated properly.'
+            logger.error(msg)
+            self._exc = TwisterRunException(msg)
+        self._thread = threading.Thread(target=self._run_simulation, args=(timeout,), daemon=True)
```

Comment:
Should we run the code here if the command is wrong? I would raise the exception in line 65: `raise TwisterRunException(msg)` to break the execution of test on setup.

Reply:
Done